coordinator/stream.go

```diff
@@ -426,29 +417,26 @@ func (s *Stream) GenerateGroupKey(ctx *streamCtx, keys []string, value *influx.R
 	if len(keys) == 0 {
 		return ""
 	}
-	builder := ctx.bp.Get()
-	defer func() {
-		builder.Reset()
-		ctx.bp.Put(builder)
-	}()
-
-	tagIndex := 0
-	for i := range keys {
-		idx := util.Search(tagIndex, len(value.Tags), func(j int) bool { return value.Tags[j].Key >= keys[i] })
-		if idx < len(value.Tags) && value.Tags[idx].Key == keys[i] {
-			builder.AppendString(value.Tags[idx].Value)
-			if i < len(keys)-1 {
-				builder.AppendByte(config.StreamGroupValueSeparator)
-			}
-			tagIndex = idx + 1
-			continue
+
+	values := make([]string, len(keys))
```

Comment:
The builder involves memory reuse. It is recommended that benchmarks be used to quantify the performance before and after the comparison.